presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -883,11 +883,24 @@ private void writeRoleGrantsFile(Set<RoleGrant> roleGrants)
 
         List<String> partitionNames = partitions.stream()
                 .map(partitionValues -> makePartitionName(table.getPartitionColumns(), ImmutableList.copyOf(partitionValues)))
+                .filter(partition -> isValidPartition(tableMetadataDirectory, partition))
                 .collect(toList());
 
         return Optional.of(ImmutableList.copyOf(partitionNames));
     }
 
+    private boolean isValidPartition(Path tableMetadataDirectory, String partitionName)
+    {
+        Path metadata = new Path(new Path(tableMetadataDirectory, partitionName), PRESTO_SCHEMA_FILE_NAME);
+
+        try {
+            return metadataFileSystem.getFileStatus(metadata).isFile();
+        }
+        catch (Exception e) {
```

Comment:
Perhaps we should be more specific and just catch `FileNotFoundException` (which according to the Javadoc is what is thrown if the path does not exist)?

Reply:
Actually it's IOException. Done